Tasks/DownloadBuildArtifactsV0/download_helper.ts

```diff
@@ -96,3 +87,47 @@ function isItemCorrupted(ticket: ArtifactDownloadTicket): boolean {
 
     return isCorrupted;
 }
+
+/**
+ * This function resolves the value for the `parallelProcessingLimit` option of `ArtifactEngine`
+ *
+ * Earlier the only way to set parallelProcessingLimit in the task
+ * was by declaring the `release.artifact.download.parallellimit` variable.
+ *
+ * To save backward compatibility we will use the following strategy:
+ *
+ * Firstly, investigate the `release.artifact.download.parallellimit` variable.
+ * If everything is okay with this variable, the task will use the value from this variable.
+ *
+ * Secondly, investigate the `Parallelization limit` input of the task.
+ * If everything is okay with the value in the related task's input, the task will use the value from this input.
+ *
+ * If validation failed for both cases the function will return the default value for the `parallelProcessingLimit` option.
+ *
+ * @param artifactDownloadLimit {string} - value of `release.artifact.download.parallellimit` variable
+ * @param taskLimit {string} - value of `Parallelization limit` task input
+ * @returns {number} - parallel processing limit
+ */
+export function resolveParallelProcessingLimit(artifactDownloadLimit: string, taskLimit: string): number {
+    const artifactDownloadParallelLimit: number = Number(artifactDownloadLimit);
+    if (!isNaN(artifactDownloadParallelLimit) && artifactDownloadParallelLimit > 0) {
+        debug('The value from "release.artifact.download.parallellimit" variable will be used in the task');
+        debug(`The parallelization limit is set to ${artifactDownloadParallelLimit}`);
+        return artifactDownloadParallelLimit;
+    } else {
+        debug('The "release.artifact.download.parallellimit" variable contains the incorrect value');
+    }
+
+    const taskInputParallelLimit: number = Number(taskLimit);
+    if (!isNaN(taskInputParallelLimit) && taskInputParallelLimit > 0) {
+        debug('The value from "Parallelization limit" input will be used in the task');
+        debug(`The parallelization limit is set to ${taskInputParallelLimit}`);
+        return taskInputParallelLimit;
+    } else {
+        debug('The "Parallelization limit" input contains the incorrect value');
+    }
+
+    debug('The parallelization limit is set to default value - 8');
+
+    return 8;
```

Comment:
Could you please move it to constants? Could it make sense to reuse it in a message above via templates?

Reply:
I moved this value to constants thanks. Thanks for the point!